### PR TITLE
RGI add parameter to save json files

### PR DESCRIPTION
**Pull request by Darcy220606**
In this PR:
- CHANGE: RGI outputs by default only `.txt` and have added the `arg_rgi_savejson` parameter to give the user the option to save it as part of the RGI output dir if required.
- REASON: We noticed that RGI outputs HUGE `.json` files, which are quite bulky and are not required downstream in the pipeline. 
-  CLOSES issue #245 

**TODO**
- [x] Usage Documentation in `docs/usage.md` is updated. NOTYET!!!
- [x] Output Documentation in `docs/output.md` is updated. NOTYET!!
- [x] `CHANGELOG.md` is updated. NOTYET!!
- [x] `README.md` is updated (including new tool citations and authors/contributors).
 
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -250,7 +250,7 @@
                     "default": "Bacteria",
                     "fa_icon": "fab fa-accusoft",
                     "description": "Specify the kingdom that the input represents.",
-                    "help_text": "Specifies the kingdom that the input sample is derived from and/or you wish to screen for\n\n> ⚠️ Prokka cannot annotate Eukaryotes.\n\nFor more information please check Prokka [documentation](https://github.com/tseemann/prokka).\n\n> Modifies tool parameter(s):\n> - Prokka: `--kingdom`",
+                    "help_text": "Specifies the kingdom that the input sample is derived from and/or you wish to screen for\n\n> \u26a0\ufe0f Prokka cannot annotate Eukaryotes.\n\nFor more information please check Prokka [documentation](https://github.com/tseemann/prokka).\n\n> Modifies tool parameter(s):\n> - Prokka: `--kingdom`",
```
Comment (louperelo) [inline nextflow_schema.json:253]: This way, the warn sign does not render on my screen on GitHub, while the old version did.

**Review by louperelo — APPROVED** [review]
LGTM

File: nextflow_schema.json (hunk 1 above)
Comment (jasmezz) [inline nextflow_schema.json:253]:
```suggestion
                    "help_text": "Specifies the kingdom that the input sample is derived from and/or you wish to screen for\n\n> ⚠️ Prokka cannot annotate Eukaryotes.\n\nFor more information please check Prokka [documentation](https://github.com/tseemann/prokka).\n\n> Modifies tool parameter(s):\n> - Prokka: `--kingdom`",
```

File: nextflow_schema.json [hunk 2]
```diff
@@ -828,7 +834,7 @@
...
                     "fa_icon": "fas fa-ruler-horizontal",
-                    "help_text": "This specifies the minimum length that the longest contig must have for the entire sample to be screened by antiSMASH.\n\nAny samples that do not reach this length will be not be sent to antiSMASH, therefore you will not receive output for these samples in your `--outdir`.\n\n> ⚠️ This is not the same as `--bgc_antismash_contigminlength`, which specifies to only analyse contigs above that threshold but _within_ a sample that has already passed `--bgc_antismash_sampleminlength` sample filter!"
+                    "help_text": "This specifies the minimum length that the longest contig must have for the entire sample to be screened by antiSMASH.\n\nAny samples that do not reach this length will be not be sent to antiSMASH, therefore you will not receive output for these samples in your `--outdir`.\n\n> \u26a0\ufe0f This is not the same as `--bgc_antismash_contigminlength`, which specifies to only analyse contigs above that threshold but _within_ a sample that has already passed `--bgc_antismash_sampleminlength` sample filter!"
```
Comment (jasmezz) [inline nextflow_schema.json:837]:
```suggestion
                    "help_text": "This specifies the minimum length that the longest contig must have for the entire sample to be screened by antiSMASH.\n\nAny samples that do not reach this length will be not be sent to antiSMASH, therefore you will not receive output for these samples in your `--outdir`.\n\n> ⚠️ This is not the same as `--bgc_antismash_contigminlength`, which specifies to only analyse contigs above that threshold but _within_ a sample that has already passed `--bgc_antismash_sampleminlength` sample filter!"
```

File: nextflow_schema.json (hunk 1 above)
Comment (jasmezz) [reply nextflow_schema.json:253]: Unfortunately, `nf-core schema build` changes this symbol into the cryptic `\u26a0\ufe0f` that is not rendered into the warning symbol... One has to change that back every time after running the schema web editor.

**Review by jasmezz — CHANGES_REQUESTED** [review]

**Review by jasmezz — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#235](https://github.com/nf-core/funcscan/pull/235) Added parameter `annotation_bakta_db_downloadtype` to be able to switch between downloading either full (33.1 GB) or light (1.3 GB excluding UPS, IPS, PSC, see parameter description) versions of the Bakta database. (by @jasmezz)
 - [#249](https://github.com/nf-core/funcscan/pull/249) Added bakta annotation to CI tests. (by @jasmezz)
 - [#251](https://github.com/nf-core/funcscan/pull/251) Added annotation tool: Pyrodigal. (by @jasmezz)
+- [#252](https://github.com/nf-core/funcscan/pull/252) Added a new parameter `-arg_rgi_savejson` that saves the file `samplename.json` in the RGI directory. The default ouput for RGI is now only `samplename.txt`. (by @darcy220606)
```
Comment (jasmezz) [inline CHANGELOG.md:14]:
```suggestion
- [#252](https://github.com/nf-core/funcscan/pull/252) Added a new parameter `-arg_rgi_savejson` that saves the file `<samplename>.json` in the RGI directory. The default ouput for RGI is now only `<samplename>.txt`. (by @darcy220606)
```

**Review by jasmezz — APPROVED** [review]
Thanks for the updates!
Why are bakta tests failing though? Also for PR #242 it looks like. Could you check if we can fix them here or in another PR (run local test and check for errors)?

File: CHANGELOG.md (hunk 3 above)
Comment (Darcy220606) [reply CHANGELOG.md:14]: @jasmezz Yup i am looking at it now

Comment (Darcy220606) [reply CHANGELOG.md:14]: BAKTA db download with `-type light` works perfectly fine locally with no errors. I guess somethings up with the ci test.

Comment (jasmezz) [reply CHANGELOG.md:14]: It shouldn't solve it, but yeah we can solve it in a different PR. Happy merging then!

**Pull request merged**